### PR TITLE
Bump golang

### PR DESCRIPTION
**Pull request by doanac**
The latest LmP is golang 1.17. Let's mark that for development to take advantages of some improvements in the language and prepare for the future deprecation of ioutil

### CODE REVIEW

File: internal/app_test.go [hunk 1]
```diff
@@ -73,11 +72,7 @@ func encrypt(t *testing.T, config map[string]*ConfigFile) {
 }
 
 func testWrapper(t *testing.T, doGet http.HandlerFunc, testFunc func(app *App, client *http.Client, tempdir string)) {
-	dir, err := ioutil.TempDir("", "")
-	if err != nil {
-		t.Error(err)
-	}
-	defer os.RemoveAll(dir)
+	dir := t.TempDir()
```
Comment (vkhoroz) [inline internal/app_test.go:75]: Does the `TempDir` auto-cleanup the directory after the last reference to it becomes inaccessible?
The `defer os.RemoveAll(dir)` is still needed if not.

**Review by vkhoroz — APPROVED** [review]
LGTM

I see you have some dependencies between your PRs... just make sure to merge them in the correct order.

File: internal/app_test.go (hunk 1 above)
Comment (doanac) [reply internal/app_test.go:75]: yep. its an easier pattern to use with unit tests

**Comment by vkhoroz** [comment]
> will rebase once more when the http PR is finished

I think I gave you a tumbsup for the HTTP Cleanup PR yesterday. Is there anything else pending?

**Pull request merged**